### DISCUSSION TITLE
Making better for determining same mapsets with metadata

### DISCUSSION THREAD

**Discussion by arken1015** [ppy/osu#20718] [Ideas]
Greeting, as mapping is also fun, I have idea for some problem which have been existed since start of osu!
Like, someone might map mapset with same mapper, same artist, same music name. But if the diffname is different it would be no problem.
BUT, if 4 things are all same(mapper name, song name, artist and difficulty name) then osu! will merge difficulty without any warning or something.

Even osu file has BeatmapID and BeatmapSetID in osu! file, it happened in recent mapping era. So I dunno if this is intended one or potential problem.

Map 1: https://osu.ppy.sh/beatmapsets/1857120#fruits/3816811
Map 2: https://osu.ppy.sh/beatmapsets/1858435#taiko/3819870
Related discussion: https://osu.ppy.sh/beatmapsets/1858435/discussion/3819870/general#/3335899

Here is the example, username 'GIGACHAD' already ranked Bring Me The Horizon - Can You Feel My Heart (Cut Ver.) as catch mapset with [GIGACHAD] difficulty. And he wanted to map Taiko mapset again. But as this issue exists, he added comma end of Taiko diff so the top diff of Taiko mapset is [GIGACHAD.] to prevent this problem.

Map 3: https://osu.ppy.sh/beatmapsets/977#osu/7717
Map 4: https://osu.ppy.sh/beatmapsets/99#osu/219

Here is another example, username 'LuigiHann' already submitted Never Gonna Give You Up and he wanted to upload again with different mapping. But as this problem exists, he added '2' end of title to prevent this problem. He commented 'Reuploading because it didn't work for some reason.'

So my idea is, if BeatmapSetID is different, then mapper would be allowed to make same song, same artist and same difficulty name. If mapset is already ranked, ignore checking duplication process. If mapset is not submitted, it would have -1 as code and it would be unique from mapset which is already submitted into BSS.

**Comment by peppy** [comment]
This has already been addressed in lazer. Are you seeing this issue on lazer or stable?

**Reply by arken1015** [reply]
Checked from Stable and got problem like this. I used https://osu.ppy.sh/beatmapsets/24010#taiko/84142 and https://osu.ppy.sh/beatmapsets/1864713#taiko/3835146 . When I download these mapsets in order like video and this problem happened. The below screenshot is when I delete 2nd mapset.

Video: https://streamable.com/b2ktik
Screenshot: https://cdn.discordapp.com/attachments/491772466566397972/1029626415630536704/unknown.png
